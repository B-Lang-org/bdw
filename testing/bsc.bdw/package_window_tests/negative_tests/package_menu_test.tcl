
# Loads the specified package
handlers::load_package Mult4

# Creates a new project
handlers::open_project project.bspec 

# Loads the specified package
handlers::load_package 

# Loads the specified package
handlers::load_package Mult4

# Reloads all currently loaded packages
handlers::reload_packages

# Shows the import hierarchy for the specified package
handlers::import_hierarchy

# Searches for the specified pattern through the package hierarchy
handlers::search_in_packages 

# Searches for the specified pattern through the package hierarchy
handlers::search_in_packages Mult -nex

# Searches for the specified pattern through the package hierarchy
handlers::search_in_packages Mu -next -previous

# Displays the project status
commands::full_clean

while {$BSPEC(FILES) == ""} {
        update
}

# Compiles the current project
handlers::compile_file Mult2.bsv -withdeps

# Waits until the process with the 'BSPEC(BUILDPID)' id has finished
while {$BSPEC(BUILDPID) != ""} {
        update
}

# Loads the specified package
handlers::load_package Mult2

# Shows the import hierarchy for the specified package
handlers::import_hierarchy Mult4

# Displays the project status
commands::full_clean

# Closes the current project
handlers::close_project

# Exits the workstation
exit
