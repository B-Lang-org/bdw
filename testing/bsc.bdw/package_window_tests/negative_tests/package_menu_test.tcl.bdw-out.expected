There is no open project
The package name is not specified
The package 'Mult4' does not exist.
There are no loaded packages.
There are no loaded packages.
The search pattern is not specified
Unknown option: -nex
Additional option is specified.
Should be  either '-next' or '-previous'.
Executing: bsc -u -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries  mult4/Mult2.bsv
WARNING: + bsc -u -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries mult4/Mult2.bsv
RESULT: checking package dependencies
RESULT: compiling mult4/Mult.bsv
RESULT: compiling mult4/Mult2.bsv
RESULT: code generation for mkMult2 starts
WARNING: Warning: "mult4/Mult2.bsv", line 23, column 8: (G0117)
WARNING:   Rule `start' shadows the effects of `cycle' when they execute in the same
WARNING:   clock cycle. Affected method calls:
WARNING:     product.write, mcand.write, mplr.write
WARNING:   To silence this warning, use the `-no-warn-action-shadowing' flag.
RESULT: Elaborated module file created: mult4/mkMult2.ba
RESULT: All packages are up to date.
Mult2 Mult PreludeBSV Prelude
The package 'Mult4' is not loaded
